elastio-s3-changelog/cloudformation-single.yaml

```
@@ -0,0 +1,263 @@
+      Example: xyz/,foo/bar/
+
+Conditions:
+  ScanExistingObjects: !Equals [!Ref ScanExistingObjects, 'true']
+  EnableQrts: !Equals [!Ref EnableQrts, 'true']
+  NoKeyPrefixes: !Equals [!Ref KeyPrefixes, '']
+
+Resources:
+  queue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 # 14 days
+      VisibilityTimeout: 43200 # 12 hours
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt dlq.Arn
+        maxReceiveCount: 3
+      Tags:
+        - Key: elastio:resource
+          Value: 'true'
+        - Key: elastio:s3-changelog-bucket
+          Value: !Ref BucketName
+        - Key: elastio:s3-changelog-stack-arn
+          Value: !Ref AWS::StackId
+
+  dlq:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600
+      VisibilityTimeout: 43200
+      Tags:
+        - Key: elastio:resource
+          Value: 'true'
+        - Key: elastio:s3-changelog-dlq
+          Value: !Ref BucketName
+        - Key: elastio:s3-changelog-stack-arn
+          Value: !Ref AWS::StackId
+
+  eventRule:
+    Type: AWS::Events::Rule
+    DependsOn: queue
+    Properties:
+      Description: !Sub Track new objects creation in ${BucketName}
+      EventPattern: !If
+        - NoKeyPrefixes
+        - !Sub >
+          {
+            "source": ["aws.s3"],
+            "detail-type": ["Object Created"],
+            "detail": { "bucket": { "name": ["${BucketName}"] } }
+          }
+        - !Join
+          - ''
+          - - !Sub >
+              {
+                "source": ["aws.s3"],
+                "detail-type": ["Object Created"],
+                "detail":
+                {
+                  "bucket": { "name": ["${BucketName}"] },
+                  "object": { "key": [
+            - !Sub
+              - '{ "prefix": "${prefixes}" }'
+              - prefixes: !Join
+                - '" },{ "prefix": "'
+                - !Split [',', !Ref KeyPrefixes]
+            - '] } } }'
+      Targets:
+        - Arn: !GetAtt queue.Arn
+          Id: changelog-queue
+          InputTransformer:
+            InputPathsMap:
+              key: '$.detail.object.key'
+              type: '$.detail-type'
+            InputTemplate:
+              '{
+                "objects": [{
+                  "key": <key>,
+                  "event_type": <type>
+                }]
+              }'
+
+  queueEventBridgePolicy:
+    Type: AWS::SQS::QueueInlinePolicy
+    Properties:
+      Queue: !Ref queue
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - sqs:SendMessage
+            Resource: !GetAtt queue.Arn
+            Principal:
+              Service: events.amazonaws.com
+            Condition:
+              ArnEquals:
+                'aws:SourceArn': !GetAtt eventRule.Arn
+
+  ssmParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: !Sub /elastio/s3/changelog/${BucketName}/${queue.QueueName}
```

I'm thinking what if we'll want to add another `s3/changelog/foo` parameter not specific to a particular queue. To make such extension possible let's add one more layer to the path
```suggestion
      Name: !Sub /elastio/s3/changelog/queue/${BucketName}/${queue.QueueName}
```